# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditorDataTransfer.cpp
# Commit: a617cdc4632e
# Full Hash: a617cdc4632e1d8c2cad4cbba28be196902ec47c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-02 21:27:32
# Regressor Bug: 1618089
# File Overlap Count: 2
# Description:
#   Bug 1618089 - part 4: Make `WSRunObject` instantiators guarantee `WSRunObject::mHTMLEditor` lifetime r=m_kato
#   
#   Some script run methods of `WSRunObject` guarantee the lifetime of its
#   `mHTMLEditor`.  However, they should be guaranteed by the instantiators
#   (all of them are `HTMLEditor`).
# ==============================================================================

diff -r ced5b6a8ab8d -r a617cdc4632e editor/libeditor/HTMLEditorDataTransfer.cpp
--- a/editor/libeditor/HTMLEditorDataTransfer.cpp	Mon Mar 02 07:35:34 2020 +0000
+++ b/editor/libeditor/HTMLEditorDataTransfer.cpp	Mon Mar 02 07:35:37 2020 +0000
@@ -372,7 +372,7 @@
   // Remove invisible `<br>` element at the point because if there is a `<br>`
   // element at end of what we paste, it will make the existing invisible
   // `<br>` element visible.
-  WSRunObject wsObj(this, pointToInsert);
+  WSRunObject wsObj(*this, pointToInsert);
   if (wsObj.GetEndReasonContent() &&
       wsObj.GetEndReasonContent()->IsHTMLElement(nsGkAtoms::br) &&
       !IsVisibleBRElement(wsObj.GetEndReasonContent())) {